c++/src/kj/table-test.c++

```diff
@@ -907,6 +908,69 @@ KJ_TEST("large tree table") {
   }
 }
 
+KJ_TEST("TreeIndex fuzz test") {
+  // A test which randomly modifies a TreeIndex to try to discover buggy state changes.
+
+  uint seed = (kj::systemPreciseCalendarClock().now() - kj::UNIX_EPOCH) / kj::NANOSECONDS;
+  KJ_CONTEXT(seed);  // print the seed if the test fails
+  srand(seed);
+
+  Table<uint, TreeIndex<UintCompare>> table;
+
+  auto randomInsert = [&]() {
+    table.upsert(rand(), [](auto&&, auto&&) {});
```

Comment:
Nit: C rand() is pretty bad. STL's [talk](https://channel9.msdn.com/Events/GoingNative/2013/rand-Considered-Harmful) on this is pretty good although I'm sure you've seen it. The main issues are that it's an extremely bad PRNG as it only generates numbers in [0, 65535] & even then it's pretty biased in its output. Probably is fine for this test, but maybe use the C++11 rand routines instead?

```
std::random_device rnd;
std::mt19937_64 eng(rnd())
std::uniform_int_distribution<uint> dist;
auto rng = [&rnd, &dist] { return dist(eng); };

uint randomNumber = rng();
```

The mersenne twister isn't the best PRNG engine (I like the PCG family more) & `random_device` probably uses `/dev/random` instead of `getrandom`, but all of this should be fine for tests.

Reply:
The C++11 interface is atrocious. Why should I have to construct three different classes with obscure names just to make a damned random number? This is an interface designed by mathematicians.

`rand()` is fine for the present use case.